src/components/TodoApp/TodoApp.tsx

```diff
@@ -0,0 +1,112 @@
+import React, { useCallback, useEffect, useState } from 'react';
+import cn from 'classnames';
+
+import { TodoList } from '../TodoList';
+import { getTodos } from '../../api/todos';
+import { useTodosContext } from '../store';
+import { useUncomplitedTodos } from '../../helpers/useUncomplitedTodos';
+import { TodoFilter } from '../TodoFilter';
+import { useComplitedTodos } from '../../helpers/useComplitedTodos';
+
+const USER_ID = 12027;
+
+export const TodoApp: React.FC = () => {
+  const { todos, reciveTodos } = useTodosContext();
+
+  const [isLoading, setIsLoading] = useState(true);
+  const [hasError, setHasError] = useState('');
+
+  const compTodosLength = useComplitedTodos().length;
+  const uncompTodosLength = useUncomplitedTodos().length;
+
+  const resetHasError = () => {
+    setHasError('');
+  };
+
+  const handlerErrors = useCallback(
+    (errMessage: string) => {
```

Comment:
It would be better to create separate enum for errors

Reply:
YEAHH, when i make review Nazar's word i saw that, and i tell what i steal that for errors 😂😂😂.

Thank you also 🤝